MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -333,23 +351,33 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel( vtkMRMLMarkups
 
   vtkSmartPointer< vtkPolyData > cleanedPolyData = vtkSmartPointer< vtkPolyData >::New();
   cleanedPolyData = cleanPointPolyData->GetOutput();
-  vtkSmartPointer< vtkPoints > cleanedPoints = vtkSmartPointer< vtkPoints >::New();
-  cleanedPoints = cleanedPolyData->GetPoints();
-  double planeNormal[ 3 ] = { 0.0, 0.0, 0.0 }; // temporary value, assigned only if points are planar
-  double lineAxis[ 3 ] = { 0.0, 0.0, 0.0 }; // temporary value, assigned only if points are linear
-  vtkSlicerMarkupsToModelLogic::PointArrangementEnum pointArrangement = ComputePointArrangement( cleanedPoints, planeNormal, lineAxis );
-  
+  // a lot of operations (vtkOBBTree, computing bounds) seem to fail on vtkPolyData, so we need vtkPoints
+  vtkPoints* cleanedPoints = cleanedPolyData->GetPoints();
+
   vtkSmartPointer< vtkDelaunay3D > delaunay = vtkSmartPointer< vtkDelaunay3D >::New();
   delaunay->SetAlpha( markupsToModelModuleNode->GetDelaunayAlpha() );
   delaunay->AlphaTrisOff();
   delaunay->AlphaLinesOff();
   delaunay->AlphaVertsOff();
+  
+  vtkSmartPointer< vtkMatrix4x4 > transformFromBoundingAxes = vtkSmartPointer< vtkMatrix4x4 >::New();
+  ComputeTransformFromBoundingAxes( cleanedPoints, transformFromBoundingAxes );
+
+  vtkSmartPointer< vtkMatrix4x4 > transformToBoundingAxes = vtkSmartPointer< vtkMatrix4x4 >::New();
+  transformToBoundingAxes->DeepCopy( transformFromBoundingAxes );
```

Comment:
Use vtkMatrix4x4::Invert(const vtkMatrix4x4 *in, vtkMatrix4x4 *out) to do it in one step.